### ISSUE TITLE
Amplify mock subscription error: "Connection failed: An invalid or illegal string was specified"

### ISSUE THREAD

**Issue by kldeb** [aws-amplify/amplify-cli#3655]
**Describe the bug**
The same code that works when subscribing to an AppSync endpoint doesn't work when using `amplify mock`.

**Amplify CLI Version**
4.13.1

**To Reproduce**
1. Create react app
2. run amplify init
3. add graphql api with api key (for test i've tried with api key and cognito user pool)
4. npm install amplify modules
5. deploy to server
6. copy paste code below 
7. run aginst server and see connection
8. run amplify mock and see error

Code: 

```js
import React, { useEffect } from 'react';
import Amplify from '@aws-amplify/core';
import API from '@aws-amplify/api';
import PubSub from '@aws-amplify/pubsub';
import config from './aws-exports';
import { onCreateMyType } from './graphql/subscriptions';

Amplify.configure(config);
PubSub.configure(config);

// Amplify.Logger.LOG_LEVEL = 'DEBUG';

function App() {
  useEffect(() => {
    let subscription;

    const run = () => {
      subscription = API.graphql({
        query: onCreateMyType,
      }).subscribe({
        next: response => console.log('response', response),
        error: response => console.log('error response', response),
      });
    };

    run();

    return () => subscription.unsubscribe();
  }, []);

  return <div>Pub Sub Test</div>;
}

export default App;

```

**Expected behavior**
Be able to test subscriptions locally 

**Desktop (please complete the following information):**
 - OS: macOS
 - Node Version: v12.10.0

**Comment by yuth** [comment]
@kldeb could you share the schema and the `onCreateMyType` subscription statements as well

**Comment by kldeb** [comment]
sure thing @yuth 

```gql
type MyType @model {
  id: ID!
  title: String!
  content: String!
  price: Int
  rating: Float
}
```

```js
export const onCreateMyType = /* GraphQL */ `
  subscription OnCreateMyType {
    onCreateMyType {
      id
      title
      content
      price
      rating
    }
  }
`;
```

**Comment by yuth** [comment]
@kldeb what version of Amplify JS package are you using. Specifically `import API from 'amplify/api` and `@aws-amplify/pubsub`.

The mock does not support Websocket based subscription yet. We should have mock support for websocket based subscription in a few weeks.

**Comment by ranjan-purbey** [comment]
Thanks @yuth  for the response. I have the same issue using `@aws-amplify/pubsub@2.1.7` and `aws-amplify@2.2.6`. As you mentioned that the mock doesn't support ws based subscription yet, is there any way to use mqtt based subscription?

**Comment by yuth** [comment]
The only way to use it would be to use older version of Amplify JS libraries (@aws-amplify/pubsub@1.x.x)

**Comment by yuth** [comment]
duplicate of https://github.com/aws-amplify/amplify-cli/issues/3008

**Comment by yuth** [comment]
this has been fixed in https://github.com/aws-amplify/amplify-cli/commit/f6dac5b6d55867e35b28b1c3eec9a6eeb4e4fbe3 and released in Amplify CLI 4.18.1
